### PR TITLE
Updates for Gazelle README

### PR DESCRIPTION
**Pull request by chrislovecnm**
- General rewrites to improve the understandability of the documentation.
- Improving the README so that the `py_test` support is more clear and more correct.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ X] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

The documentation needs some wordsmithing, and the instructions for `py_test` targets are incorrect.

Issue Number: N/A

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@f0rmiga @alexeagle PTAL

**Comment by chrislovecnm** [comment]
@aignas thoughts?

**Comment by chrislovecnm** [comment]
I'm getting an odd error from CI/CD on macos

```
(21:12:56) ERROR: /Users/buildkite/builds/bk-imacpro-7/bazel/rules-python-python/examples/pip_parse/BUILD.bazel:59:25: Middleman _middlemen/requirements_Utest-runfiles failed: missing input file '@pypi__setuptools//:setuptools/command/__pycache__/bdist_egg.cpython-39.pyc.4385037200'
 ```

File: gazelle/README.md [hunk 1]
```diff
@@ -143,14 +142,12 @@ Python-specific directives are as follows:
 | Controls the `py_library` naming convention. It interpolates $package_name$ with the Bazel package name. E.g. if the Bazel package name is `foo`, setting this to `$package_name$_my_lib` would result in a generated target named `foo_my_lib`. | |
 | `# gazelle:python_binary_naming_convention` | `$package_name$_bin` |
 | Controls the `py_binary` naming convention. Follows the same interpolation rules as `python_library_naming_convention`. | |
-| `# gazelle:python_test_naming_convention` | `$package_name$_test` |
```
Comment (f0rmiga) [inline gazelle/README.md:146]: Why are we removing this?

File: gazelle/README.md [hunk 2]
```diff
@@ -166,27 +163,52 @@ dependencies are added to the `deps` attribute.
...
+py_test(
+    name = "build_file_generation_test",
+    srcs = ["__test__.py"],
+    main = "__test__.py",
+    deps = [":build_file_generation"],
+)
+```
+
+Often Python unit test files are named with the prefix `test_` or the suffix `_test`.
+At this point, the plugin does not support generating `py_test` targets for files
```
Comment (f0rmiga) [inline gazelle/README.md:180]: We merged a change that will generate individual test targets for each file matching this pattern, but only if a `__test__` target is not present.

**Review by f0rmiga — CHANGES_REQUESTED** [review]

File: gazelle/README.md (hunk 2 above)
Comment (chrislovecnm) [reply gazelle/README.md:180]: When was it merged?  Cause this is what I could never get to work :(

Comment (f0rmiga) [reply gazelle/README.md:180]: https://github.com/bazelbuild/rules_python/commit/0943375cbbef6b1f73c439b5e350ac93b1b22bfa

**Comment by chrislovecnm** [comment]
Closing, as I am including these updates in a new PR.

**Pull request closed without merging**